GVFS/GVFS.Common/Git/DiffTreeResult.cs

```diff
@@ -84,6 +86,8 @@ public static DiffTreeResult ParseFromDiffTreeLine(string line, string repoRoot)
             DiffTreeResult result = new DiffTreeResult();
             result.SourceIsDirectory = ValidTreeModes.Contains(parts[0]);
             result.TargetIsDirectory = ValidTreeModes.Contains(parts[1]);
+            result.SourceMode = parts[0];
+            result.TargetMode = parts[1];
 
             if (!result.TargetIsDirectory)
             {

```

Comment:
This line:

`result.TargetIsSymLink = parts[1] == "120000";`

Can now be changed to:

`result.TargetIsSymLink = result.TargetMode == "120000";`

Which (I think) is more readable.

Reply:
This code is now a little bit stranger because of the ushort conversions, but if you have a good suggestion on how to make this cleaner, I'd love to take it.

Reply:
Maybe something like:

```
private const ushort SymLinkFileIndexEntry = 0xA000;

result.TargetIsSymLink = result.TargetMode == SymLinkFileIndexEntry;
```

Note that SymLinkFileIndexEntry is already in `GitIndexProjection.FileTypeAndMode`, and so you could move it from there to a common location .

Reply:
You could do the same check below at line 159 as well:

`blobAdd.TargetIsSymLink = line.StartsWith("120000");`